rest/utilities_testing.go

```diff
@@ -814,3 +1046,95 @@ func EnableBlipSyncLogs() {
 	base.EnableLogKey("Sync")
 	base.EnableLogKey("Sync+")
 }
+
+
+type AttachmentMap map[string]*DocAttachment
+
+type DocAttachment struct {
+	ContentType string `json:"content_type,omitempty"`
+	Digest      string `json:"digest,omitempty"`
+	Length      int    `json:"length,omitempty"`
+	Revpos      int    `json:"revpos,omitempty"`
+	Stub        bool   `json:"stub,omitempty"`
+	Data        []byte `json:-` // tell json marshal/unmarshal to ignore this field
+}
+
+// --------------------------------------------
+
```

Comment:
Maybe some comments to explain what RestDocument is, and why it's only needed on the test side.  

If it's not being used in non-test scenarios, I assume that means any future changes to these internal document properties will require double-fixing (here and in the code).

Reply:
> If it's not being used in non-test scenarios

It's only currently being used in the test scenarios and their helpers, but I was trying to come up with something generic that would be useful across the codebase.

I'll add some comments.